include/my_dbug.h

```diff
@@ -209,16 +209,6 @@ void debug_sync_point(const char* lock_name, uint lock_timeout);
 
 #ifdef __cplusplus
 }
-# ifdef DBUG_OFF
-#  define DBUG_LOG(keyword, v) do {} while (0)
-# else
-#  include <sstream>
```

Comment:
I guess this line is culprit. I kind of like C++ streams and I would prefer to keep `DBUG_LOG()` here.
It compiles pretty well if we just remove this line. And add a comment:
```
/*
  DBUG_LOG() was initially intended for InnoDB. To be able to use it elsewhere
  one should #include <sstream>. We intentially avoid including it here to save
  compilation time.
*/
```

I'd suggest to fix this in 10.2.

Reply:
I could cherry-pick to 10.2 myself though.

Reply:
I've tried to factor out a function and use `<iosfwd>` here but failed.

I'll close this PR and create a new one for `10.2`.

Reply:
Btw, you cold like [this library](http://fmtlib.net) better than standard streams.

Reply:
I didn't really test it, but per my understanding C++ stream doesn't have to parse format string. Of course it needs to convert numbers to strings, etc. But otherwise it should be no different from `memcpy()`.

This library claims it is faster than C++ stream, while it still has to parse format string.

Reply:
Yes it does. But who cares if it's performance is better?

Format strings are less verbose and easy to read than 'chevron hell'. And it's stateless and you don't need stuff [like that](http://www.boost.org/doc/libs/1_64_0/libs/io/doc/ios_state.html). Also, 'libfmt' is proposed to standard.

Reply:
Yeah, sure. I'm just trying to understand why is it faster taking into account it has to do more work?

Reply:
I don't know a simple answer, sorry. I've never profiled it by myself.